modules/client/src/main/scala/ch/epfl/bluebrain/nexus/admin/client/AdminClient.scala

```diff
@@ -0,0 +1,104 @@
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.stream.Materializer
+import ch.epfl.bluebrain.nexus.admin.client.config.AdminConfig
+import ch.epfl.bluebrain.nexus.admin.client.types.Project
+import ch.epfl.bluebrain.nexus.admin.refined.project.ProjectReference
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient.UntypedHttpClient
+import ch.epfl.bluebrain.nexus.commons.http.JsonLdCirceSupport.unmarshaller
+import ch.epfl.bluebrain.nexus.commons.http.{HttpClient, UnexpectedUnsuccessfulHttpResponse}
+import ch.epfl.bluebrain.nexus.commons.iam.acls.FullAccessControlList
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity
+import ch.epfl.bluebrain.nexus.commons.iam.io.serialization.JsonLdSerialization
+import ch.epfl.bluebrain.nexus.commons.types.HttpRejection.UnauthorizedAccess
+import io.circe.Decoder
+import io.circe.generic.auto._
+import journal.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Admin client contract
+  *
+  * @tparam F the monadic effect type
+  */
+trait AdminClient[F[_]] {
+
+  /**
+    * Retrieves a [[Project]] resource instance.
+    *
+    * @param name        the project name
+    * @param credentials the optionally provided [[OAuth2BearerToken]]
+    */
+  def getProject(name: ProjectReference, credentials: Option[OAuth2BearerToken]): F[Project]
+
+  /**
+    * Retrieves ACLs for a given project resource instance.
+    *
+    * @param name        the project name
+    * @param credentials the optionally provided [[OAuth2BearerToken]]
+    */
+  def getProjectAcls(name: ProjectReference, credentials: Option[OAuth2BearerToken]): F[FullAccessControlList]
+
+}
+
+object AdminClient {
+
+  private val log = Logger[this.type]
+
+  /**
+    * Builds an [[AdminClient]] instance from a provided configuration and implicitly available instances of
+    * [[ExecutionContext]], [[Materializer]] and [[UntypedHttpClient]].
+    *
+    * @param config the provided [[AdminConfig]] instance
+    */
+  def apply(config: AdminConfig)(implicit
+                                 ec: ExecutionContext,
+                                 mt: Materializer,
+                                 cl: UntypedHttpClient[Future]): AdminClient[Future] = {
+    implicit val identityDecoder: Decoder[Identity] = JsonLdSerialization.identityDecoder
+
+    val projectClient = HttpClient.withAkkaUnmarshaller[Project]
+    val aclsClient    = HttpClient.withAkkaUnmarshaller[FullAccessControlList]
+
+    new AdminClient[Future] {
+
+      override def getProject(name: ProjectReference, credentials: Option[OAuth2BearerToken]): Future[Project] = {
+        val path = Path(name.value)
+        projectClient(requestFrom(credentials, path)).recoverWith { case e => recover(e, path) }
+      }
+
+      override def getProjectAcls(name: ProjectReference,
+                                  credentials: Option[OAuth2BearerToken]): Future[FullAccessControlList] = {
+        val path = Path(name.value) + "/acls"
```

Comment:
I think you can use the Path constructor like this: `name.value / "acls"` if you import the Path syntax